__tests__/app.test.js

```diff
@@ -36,17 +44,82 @@ describe("GET/api/articles/:article_id", () => {
     return request(app)
       .get("/api/articles/1")
       .expect(200)
-      .then(({ body }) => {
-        expect.objectContaining(body);
+      .then(({ body: article }) => {
+        expect(article).toHaveLength(1);
+        article.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              article_id: expect.any(Number),
+              title: expect.any(String),
+              topic: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number)
+            })
+          );
+        });
       });
   });
-  test("404:article id not in database", () => {
+  test("400:article id not in database", () => {
     return request(app)
       .get("/api/articles/1123")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Id not found" );
+      });
+  });
+});
+
+describe("GET/api/users/", () => {
+  test("200: return users", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body: users }) => {
+        expect(users).toHaveLength(4);
+        users.forEach((users) => {
+          expect(users).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String)
+            })
+          );
+        });
+      });
+  });
+  test("404: mispelt endpoint", () => {
```

Comment:
This only needs to be tested once for the whole app. It will cover any endpoints that we have not designed a path for. Therefore, your test on line 32 will cover the same thing (although in that test you should be checking that the user receives a response with the expected message, not just an empty body).